xdsl/dialects/builtin.py

```diff
@@ -646,6 +646,11 @@ def format(self) -> str:
 AnySignlessIntegerType: TypeAlias = Annotated[IntegerType, SignlessIntegerConstraint]
 """Type alias constrained to signless IntegerType."""
 
+_IntegerType = TypeVar(
+    "_IntegerType", bound=IntegerType, covariant=True, default=IntegerType
+)
+_IntegerTypeInvT = TypeVar("_IntegerTypeInvT", bound=IntegerType)
```

Comment:
```suggestion
_IntegerTypeInvT = TypeVar("_IntegerTypeInvT", bound=IntegerType, default=IntegerType)
```
No reason not to include a default I think